### PR TITLE
Deprecate Show

### PR DESCRIPTION
**Pull request by nomisRev**
This PR deprecates `Show` since Kotlin, and Kotlin's Std doesn't take `Show` into account nor does Kotlin have the power to automatically derive or inject `Show` anywhere. This makes `Show` very cumbersome to work with, see [Tuple10Show.kt](https://github.com/arrow-kt/arrow-core/compare/sv-deprecate-show?expand=1#diff-7f26da547470ab066e076a5795cdd1e670a17d5eded730b20bf9a7ecd844be86).

The use-cases of `Show` seemed very limited and niche to us, so we've decided to exclude it from Arrow in 1.0.0. The common technique for this is to use `data class`, which automatically implements `toString()` and correctly implementing `toString()`.

### CODE REVIEW

File: arrow-core-data/src/main/kotlin/arrow/core/Either.kt [hunk 1]
```diff
@@ -1078,20 +1078,21 @@ sealed class Either<out A, out B> : EitherOf<A, B> {
...
 
-    override fun toString(): String = show(Show.any(), Show.any())
+    override fun toString(): String = "Either.Right($b)"
 
     companion object {
       operator fun <B> invoke(b: B): Either<Nothing, B> = Right(b)
     }
   }
 
   fun show(SL: Show<A>, SR: Show<B>): String = fold(
```
Comment (franciscodr) [inline arrow-core-data/src/main/kotlin/arrow/core/Either.kt:1088]:
```suggestion
  @Deprecated(ShowDeprecation)
  fun show(SL: Show<A>, SR: Show<B>): String = fold(
```

File: arrow-core-data/src/main/kotlin/arrow/core/Eval.kt [hunk 2]
```diff
@@ -532,6 +544,9 @@ sealed class Eval<out A> : EvalOf<A> {
     abstract fun <S> run(s: S): Eval<A>
     override fun memoize(): Eval<A> = Memoize(this)
     override fun value(): A = evaluate(this)
+
+    override fun toString(): String =
+      "Eval.FlatMao(..)"
```
Comment (franciscodr) [inline arrow-core-data/src/main/kotlin/arrow/core/Eval.kt:549]:
```suggestion
      "Eval.FlatMap(..)"
```

File: arrow-core-data/src/main/kotlin/arrow/core/Eval.kt [hunk 3]
```diff
@@ -547,7 +562,13 @@ sealed class Eval<out A> : EvalOf<A> {
     override fun value(): A = result.getOrElse {
       evaluate(eval).also { result = Some(it) }
     }
+
+    override fun toString(): String =
+      "Eval.Memoize($eval)"
   }
+
+  override fun toString(): String =
```
Comment (franciscodr) [inline arrow-core-data/src/main/kotlin/arrow/core/Eval.kt:570]: I was wondering why we need the `toString` method in the sealed class

Comment (nomisRev) [reply arrow-core-data/src/main/kotlin/arrow/core/Eval.kt:570]: I guess it's not needed if it's defined for all the cases.

**Comment by franciscodr** [comment]
I think we need to remove the instance of Show for Validated and its implementation:
- [This line](https://github.com/arrow-kt/arrow-core/pull/321/files#diff-458531809e0125ff24d909bd1213c7e01ef68ecdbbff1537a1a13e4e163cbf3cR961)
- And [this line](https://github.com/arrow-kt/arrow-core/pull/321/files#diff-458531809e0125ff24d909bd1213c7e01ef68ecdbbff1537a1a13e4e163cbf3cR1270)

**Review by franciscodr — APPROVED** [review]
Thanks, @nomisRev!

**Review by i-walker — APPROVED** [review]
Thanks @nomisRev Looks great

**Pull request merged**